src/components/TradeAssetSearch/hooks/useGetCustomTokensQuery.tsx

```diff
@@ -42,18 +48,50 @@ export const useGetCustomTokensQuery = ({
     [contractAddress],
   )
 
+  const getSolanaTokenMetadata = useCallback(
+    async (mintAddress: string): Promise<TokenMetadata> => {
+      const solanaRpcUrl = `${getConfig().REACT_APP_ALCHEMY_SOLANA_BASE_URL}/${
+        getConfig().REACT_APP_ALCHEMY_API_KEY
+      }`
+      const connection = new Connection(solanaRpcUrl)
+      const metaplex = Metaplex.make(connection)
+      const metadata = await metaplex.nfts().findByMint({ mintAddress: new PublicKey(mintAddress) })
+
+      return {
+        name: metadata.name,
+        symbol: metadata.symbol,
+        decimals: metadata.mint.currency.decimals,
+        chainId: solanaChainId,
+        contractAddress: mintAddress,
+        price: '0',
+        logo: metadata.json?.image ?? '',
+      }
+    },
+    [],
+  )
+
   const isValidEvmAddress = useMemo(
     () => isAddress(contractAddress, { strict: false }),
     [contractAddress],
   )
 
   const getQueryFn = useCallback(
-    (chainId: ChainId) => () => getTokenMetadata(chainId),
-    [getTokenMetadata],
+    (chainId: ChainId) => () => {
+      if (chainId === solanaChainId) {
+        return getSolanaTokenMetadata(contractAddress)
+      } else if (isValidEvmAddress) {
+        return getEvmTokenMetadata(chainId)
+      } else {
+        return undefined
```

Comment:
[nitpick]: Can actually negate the requirement for the termary on `shouldQuery` below by returning skipToken here

```suggestion
        return skipToken
```

Reply:
https://github.com/shapeshift/web/pull/8619/commits/68c9e166df4d148d2ffa9027bf9ca0226831da7c